### PR TITLE
Move some diagnostic-relevant methods from InferCtxt to TypeErrCtxt

### PR DESCRIPTION
**Pull request by compiler-errors**
Self-explanatory. IDK why these weren't moved onto `TypeErrCtxt` when that struct was created? Maybe just an oversight.

cc #127492

r? lcnr

### CODE REVIEW

File: compiler/rustc_infer/src/infer/error_reporting/mod.rs [hunk 1]
```diff
@@ -398,7 +398,7 @@ pub fn unexpected_hidden_region_diagnostic<'a, 'tcx>(
     err
 }
 
-impl<'tcx> InferCtxt<'tcx> {
+impl<'tcx> TypeErrCtxt<'_, 'tcx> {
```
Comment (lcnr) [inline compiler/rustc_infer/src/infer/error_reporting/mod.rs:401]: can merge the impl blocks here

File: compiler/rustc_infer/src/infer/error_reporting/mod.rs [hunk 2]
```diff
@@ -2792,7 +2792,7 @@ impl<'tcx> TypeRelation<TyCtxt<'tcx>> for SameTypeModuloInfer<'_, 'tcx> {
     }
 }
 
-impl<'tcx> InferCtxt<'tcx> {
```
Comment (lcnr) [inline compiler/rustc_infer/src/infer/error_reporting/mod.rs:2795]: same here :thinking:

**Review by lcnr — COMMENTED** [review]
r=me after nit

**Comment by compiler-errors** [comment]
This is impossible to rebase!

**Pull request closed without merging**